[Plugin] styled-bars

Araxeus opened the pull request:
I got bored of the old bars :)
This plugin changes the Titlebar, main menu, and scrollbar

![23-03-2021 02 54 23](https://user-images.githubusercontent.com/78568641/112076657-80a1dc00-8b83-11eb-9782-6fcee3ed7df7.png)

th-ch reviewed (commented):
Thanks for the contribution, love the idea! 🤩 

Left a few comments (main one is about re-using the existing menu to avoid duplicating code, otherwise it will make maintenance of the code harder - also, IMHO `styled-bars` feels a bit unclear as plugin name, I would have gone with something explicit like `in-app-menu`) - I am fine with merging it as it is and make the follow ups, or you can update the PR if you feel like it! It's up to you, in both cases this contribution is already awesome 👍 

Note: it does not work for me on MacOS but it might be the same issue as https://github.com/AlexTorresSk/custom-electron-titlebar/issues/145 (from the package)

Araxeus commented on plugins/styled-bars/front.js:
```diff
@@ -0,0 +1,23 @@
+			myBar.updateMenu(remote.Menu.getApplicationMenu());
+		} else {
+			try {
+				myBar.updateMenu(null);
```
I dont want to dispose the menu because I will have to recreate it from scratch instead of updating it

Araxeus commented on plugins/styled-bars/back.js:
```diff
@@ -0,0 +1,300 @@
+const {getAllPlugins} = require('../../plugins/utils');
+const config = require('../../config');
+
+const pluginEnabledMenu = (plugin, label = "") => ({
```
Done :D

Araxeus commented:
You are welcome to change the name to whatever you like :)
It was `styled-titlebar` at the beginning , but then I added the scrollbar so I changed the name 😂

It was tested only on windows but from what i've seen - mac users can use their own special titlebar ;)

About the comments on the code
- I changed the menu logic
It nows only fixes the existing template instead of creating a new one. 
Seems to work great for now but i will test drive it more

Also is it ok if I integrate a folder chooser for the `downloader` plugin if you aren't planning on doing that?

(to plugins/downloader/menu.js)

Araxeus commented:
I also added to the main menu a change that i had in my copied template :
Quit button + Restart button
I put them in the view submenu for now
Of course this has nothing to do with the plugin, you are welcome to disregard that commit if you want.
https://github.com/th-ch/youtube-music/pull/201/commits/09d2feb15bdcfaadeb94a648d1980088b8245cc9
